R/data.table.R

```diff
@@ -200,6 +200,21 @@ replace_dot_alias <- function(e) {
   x
 }
 
+.checkTypos = function(err, ref) {
+  if (grepl('object.*not found', err$message)) {
+    used = gsub(".*object '([^']+)'.*", "\\1", err$message)
+    found = agrep(used, ref, value = TRUE, ignore.case = TRUE, fixed = TRUE)
+    if (length(found)) {
+      stop("Object '", used, "' not found. Perhaps you ",
+           sprintf("intended one of: [%s]",
+                   paste(found, collapse = ', ')))
```

Comment:
2 suggestions:
1. Add a special case for `length(found) == 1` ("one of ..." seems awkward for 1 item).
2. Limit the number of columns suggested to 3 or 4 (in theory your DT can have 1000s of columns, similarly named, and all of them might be found by `agrep`).